apps/charterafrica/src/lib/data/common/blockify/articles.js

```diff
@@ -0,0 +1,203 @@
+import articlesQueryString from "@/charterafrica/utils/articles/queryString";
+import formatDateTime from "@/charterafrica/utils/formatDate";
+
+export async function getTags(page, api, context) {
+  const { locale } = context;
+  const { slug: collection } = page;
+  const { docs } = await api.getCollection(collection, {
+    locale,
+    where: {
+      _status: { equals: "published" },
+    },
+  });
+
+  if (!docs?.length) {
+    return null;
+  }
+  const tags = docs.flatMap((doc) => doc.tags).filter((tag) => tag?.slug);
```

Comment:
using typescript might help a lot with unsafe type issues here.

Reply:
💯 ... that's the next step: monorepo -> ci/cd -> typescript -> end-to-end testing. Happy to hear your suggestions as well.

Reply:
you mean we are going to convert our codebase based on this steps

Reply:
Yes-ish.

1. We've moved about 3 apps into this monorepo (charter.africa, codeforafrica and promisetracker). We still have a few to go (pesayetu, trolltracker, etc.)
2. We've already implemented CI/CD for charter.africa and codeforafrica (you can see the workflows in the repo).
3. We have included Jest (for component testing) and Playwright (for end-to-end testing) but we're not really using them that much (just snapshot testing w/ Jest).

So the idea is come up with a comprehensive plan that will allow us to move everything UI into this repo and make sure we have ci/cd -> typescript -> end-to-end testing for all the apps.